src/Psalm/Internal/Type/Comparator/ScalarTypeComparator.php

```diff
@@ -89,9 +89,7 @@ public static function isContainedBy(
         }
 
         if ($container_type_part instanceof TNonspecificLiteralString
-            && ($input_type_part instanceof TLiteralString
-                || $input_type_part instanceof TNonspecificLiteralString
-                || $input_type_part instanceof TNonEmptyNonspecificLiteralString)
```

Comment:
TNonEmptyNonspecificLiteralString is a child of TNonspecificLiteralString just above